src/GitHub.Exports/Services/EnterpriseProbeTask.cs

```diff
@@ -37,12 +37,18 @@ public EnterpriseProbeTask(IProgram program, IHttpClient httpClient)
             };
             request.Headers.Add("User-Agent", productHeader.ToString());
 
+            var success = false;
             var ret = await httpClient
                     .Send(request, CancellationToken.None)
-                    .Catch(ex => null);
+                    .Catch(ex => {
+                        var apiex = ex as ApiException;
+                        if (apiex != null)
+                            success = apiex.HttpResponse?.Headers.ContainsKey("X-GitHub-Request-Id") ?? false;
```

Comment:
Or even better, but this check in a method in a common location and call it in both places. Something like: `IsGitHubApiException(Exception exception);`